File: src/test/java/io/lsdconsulting/lsd/distributed/integration/testapp/config/RepositoryConfig.java
package io.lsdconsulting.lsd.distributed.integration.testapp.config;

import io.lsdconsulting.lsd.distributed.integration.testapp.repository.TestRepository;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;

import static io.lsdconsulting.lsd.distributed.integration.testapp.repository.TestRepository.setupDatabase;

@TestConfiguration
public class RepositoryConfig {

    // This is because the configs in spring.factories run always before any test configs.
    static {
        setupDatabase();
    }

    @Bean
    public TestRepository testRepository() {
        return new TestRepository();
    }
}

File: integrationMongoTest/src/test/kotlin/io/lsdconsulting/lsd/distributed/interceptor/integration/testapp/config/RepositoryConfig.kt
package io.lsdconsulting.lsd.distributed.interceptor.integration.testapp.config

import io.lsdconsulting.lsd.distributed.interceptor.integration.testapp.repository.TestRepository
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean

@TestConfiguration
open class RepositoryConfig {

    @Bean
    open fun testRepository() = TestRepository()
}
